frontend/packages/console-dynamic-plugin-sdk/src/app/modal-support/ModalProvider.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import { AsyncLoader } from '../../utils/async';
+
+type CloseModal = () => void;
+
+export type LaunchModal = (
+  component: () => Promise<React.FC<{ closeModal: CloseModal }>>,
```

Comment:
Why impose an async loading component instead of just a component and let the consumer choose whether or not to use an async component.

Maybe `launchModal` can be simplified to a render function that will return the actual JSX for the modal.
```
export type LaunchModal = (({ closeModal: CloseModal }) => ReactElement) => void;
```

Reply:
ack